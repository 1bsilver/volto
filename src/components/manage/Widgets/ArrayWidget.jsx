/**
 * ArrayWidget component.
 * @module components/manage/Widgets/ArrayWidget
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Form, Label, TextArea } from 'semantic-ui-react';
import { map } from 'lodash';

/**
 * ArrayWidget component class.
 * @function ArrayWidget
 * @returns {string} Markup of the component.
 */
const ArrayWidget = ({
  id,
  title,
  required,
  description,
  error,
  value,
  onChange,
}) => (
  <Form.Field required={required} error={error.length > 0}>
    <label htmlFor={`field-${id}`}>
      {title}
      {description && <span className="help">{description}</span>}
    </label>
    <TextArea
      id={`field-${id}`}
      name={id}
      value={value ? value.join('\n') : ''}
      onChange={({ target }) =>
        onChange(
          id,
          target.value === '' ? undefined : target.value.split('\n'),
        )}
    />
    {map(error, message => (
      <Label key={message} basic color="red" pointing>{message}</Label>
    ))}
  </Form.Field>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
ArrayWidget.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  required: PropTypes.bool,
  error: PropTypes.arrayOf(PropTypes.string),
  value: PropTypes.arrayOf(PropTypes.string),
  onChange: PropTypes.func.isRequired,
};

/**
 * Default properties.
 * @property {Object} defaultProps Default properties.
 * @static
 */
ArrayWidget.defaultProps = {
  description: null,
  required: false,
  error: [],
  value: null,
};

export default ArrayWidget;
