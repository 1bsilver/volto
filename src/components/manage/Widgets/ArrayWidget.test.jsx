import React from 'react';
import renderer from 'react-test-renderer';
import ArrayWidget from './ArrayWidget';

test('renders an array widget component', () => {
  const component = renderer.create(
    <ArrayWidget
      id="my-field"
      title="My field"
      onChange={() => {}}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
