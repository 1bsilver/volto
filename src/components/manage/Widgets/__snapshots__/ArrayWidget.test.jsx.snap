// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an array widget component 1`] = `
<div
  className="field"
>
  <label
    htmlFor="field-my-field"
  >
    My field
  </label>
  <textarea
    id="field-my-field"
    name="my-field"
    onChange={[Function]}
    onInput={[Function]}
    rows={3}
    style={
      Object {
        "resize": "",
      }
    }
    value=""
  />
</div>
`;
